CONTRIBUTING.md

```diff
@@ -0,0 +1,244 @@
+```
+This allows everybody to understand how your code improved Braindecode from the last version! :)
+The ```:gh:`198` ``` part refers to the corresponding Pull Request which we will get to further below. 
+You may leave it out for now.
+Don't forget to add yourself to the list of authors at the end of the file, for example like this:
+```
+.. _Hubert Banville: https://github.com/hubertjb
+```
+
+##### Add your name to code file author list
+Also ensure to add your name at the top of any code file you edited, for example like:
+```
+# Authors: Hubert Banville <hubert.jbanville@gmail.com>
+[...]
+```
+
+##### Add an example file
+If you add functionality that could be nicely explained with a code example, add a file to the `examples` folder with a `plot_` at the beginning of the file name.
+For example check `examples/plot_bcic_iv_2a_moabb_trial.py`. 
+These files will be executed when building of the documentation.
+
+##### Build and check the documentation
+As explained above, you can run
+ ```
+cd docs
+make html
+```
+and check the built documentation under `docs/_build/html/index.html`.
+
+### Commit and push to your fork
+Now you can add, commit and push to your local fork.
+For example, if you had modified `datautil/xy.py` following all the steps above, it may look like this:
+```
+git add datautil/xy.py test/unit_tests/datautil/test_xy.py docs/api.rst docs/whats_new.rst
+git commit -m '<yourcommitmessage>'
+git push -u origin <branch-name>
+```
+
+You can use `git diff` and `git status` to see if you missed any changes you made.
+You have to repeat these steps once you change your code further and want to push it again.
+
+### Start a Pull Request
+
+Open your fork on https://github.com, after you pushed, there should be a button to make a pull request to the main braindecode repository from your new branch.
+Click that button and describe a bit more what you have changed so others can review your changes.
+Now you should see your pull request appearing under https://github.com/braindecode/braindecode/pulls. The continuous integration will try to run the tests and built the documentation from your pull request.
+You can also later add the number of your pull request to `docs/whats_new.rst`.
+
+Congratulations, now you have made your first Braindecode pull request!
+In order to keep your pull request up to date when you have made further changes, simply rerun add/commit/push as above.
+
+
+### Rebasing from upstream master
+If you worked for a longer time in your branch and the upstream Braindecode master branch has already changed a lot,
+it may become necessary to rebase your code changes on the current Braindecode master.
+This will make the git history look as if you just started your changes from the current upstream master state.
+To do that, from your local repository, first commit all the changes inside your local branch (without pushing) and then run the following:
+```
+git checkout master
```

Comment:
You can replace checking out to master and back as well as pulling master with this line: 
`git fetch upstream master:master`
It will make your master equal to the upstream one without changing branches.

Reply:
Alternatively you can do:
```
git fetch upstream
git rebase upstream/master
```

Reply:
Great thanks! changed it, I did leave the checkout <branch-name> inside to prevent accidental rebases to other branch